### PR TITLE
[azopenai] Fixing incorrect type for MongoDB auth

### PR DESCRIPTION
**Pull request by richardpark-msft**

### CODE REVIEW

**Comment by richardpark-msft** [comment]
@glecaros, just doing this quick fix, but no need to re-release. It just corrects an issue with the type, but there's a simple workaround.

File: sdk/ai/azopenai/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### Bugs Fixed
 
+- `MongoDBChatExtensionParameters.Authentication`'s type has been changed to a `OnYourDataUsernameAndPasswordAuthenticationOptions`. (PR#23620)
```
Comment (jhendrixMSFT) [inline sdk/ai/azopenai/CHANGELOG.md:11]: The current change is breaking, so this should move to the applicable section.

Comment (richardpark-msft) [reply sdk/ai/azopenai/CHANGELOG.md:11]: Oof, you're exactly right. Moving...

**Review by richardpark-msft — COMMENTED** [review]
/azp run go - azopenai

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by richardpark-msft** [comment]
@microsoft-github-policy-service rerun

**Pull request merged**